### PR TITLE
Saas 12.2 fix l10n country rli

### PR DESCRIPTION
**Pull request by RomainLibert**
Writing the country_id of the main company in demo data disables demo
data installation if installing multiple l10n, this is caused by a wrong
check of the vat field (base_vat).

Two approaches can be used here:

  - Always write on country_id in all l10n demo data
    NB: this also implies writing the country_id first in order for the
        vat check to be done using the right country code
  - Never write on country_id in any l10n demo data

The prefered approach for the moment is to never write on the country_id
of the main company.
This approach is easier to apply as it is a mimimal diff and the first
approach would require deeper changes to the code in order to allow
writing all the information in a single `record` tag in the demo data

### CODE REVIEW

File: addons/base_vat/models/res_partner.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ def fix_eu_vat_number(self, country_id, vat):
                 vat = country_code + vat
         return vat
 
-    @api.constrains('vat')
+    @api.constrains('vat', 'commercial_partner_id.country_id.code')
```
Comment (nhomar) [inline addons/base_vat/models/res_partner.py]: isn't just `commercial_partner_id.country_id` enough?

Comment (RomainLibert) [reply addons/base_vat/models/res_partner.py]: In fact country_id should be enough

**Comment by d-fence** [comment]
@robodoo r+

File: addons/base_vat/models/res_partner.py (hunk 1 above)
Comment (sswapnesh) [reply addons/base_vat/models/res_partner.py]: > isn't just `commercial_partner_id.country_id` enough?

Just for Info: 

`@constrains only supports simple field names, dotted names (fields of relational fields e.g. partner_id.customer) are not supported and will be ignored`
See: https://github.com/odoo/odoo/blob/12.0/odoo/api.py#L156

**Comment by d-fence** [comment]
robodoo rebase-ff

**Pull request closed without merging**